### ISSUE TITLE
Status messages not auto updating

### ISSUE THREAD

**Issue by aeonoea** [NooberPro/minecraft-discord-bot#19]
Hi, great job with this project! So we're at the point with configuration where pretty much everything works, except the status messages that are posted just once and do not update. We tried both with '/status' and "/setstatus" and it doesn't make a difference. Server and port are configured in the config.js. Any idea why this could be?

**Comment by aeonoea** [comment]
Thanks for your offer to help! I think server side the API is working as it should, since the bot's status updates are continually updating (every 60 seconds, in the top right activity tab). The feature that doesn't seem to be working is the status messages in the channel. It only shows the current server status once and then doesn't update anymore. I can post more details tomorrow and also the config.js for you to have a look.

**Comment by NooberPro** [comment]
Hey, thanks for letting me know! I'll take a closer look into this and see what's going on. Feel free to share the config.js whenever you're ready.

**Comment by aeonoea** [comment]
So here's the config. Let me know if you need more information or logs, etc.

`// Config Documentation: https://nooberpro.gitbook.io/minecraft-discord-bot/installation/config
// "MC" refers to Minecraft in the comments for convenience.
module.exports = {
  bot: {
    token: 'xxx',
    // Automatically updates the bot's status and activity.
    presence: {
      enabled: true,
      activity: 'Playing', // Options: Playing, Listening, Watching, Competing.
      text: {
        online: 'with {playeronline}/{playermax} players', // {playeronline} and {playermax} display the current and maximum number of players.
        offline: 'Server Offline', // Status text when the server is offline.
      },
      status: {
        // Options: online, idle, dnd, invisible.
        online: 'online', // Bot status when the MC server is online.
        offline: 'idle', // Bot status when the MC server is offline.
      },
    },
  },
  mcserver: {
    ip: 'xxx', // IP address of the MC server.
    port: 25568, // Port number of the MC server. Use Query Port in Java for the full player list.
    type: 'java', // Type of MC server: "java" or "bedrock".
    name: 'xxx', // Name of the MC server.
    version: 'Purpur 1.21', // Version of the MC server.
    icon: 'xxx', // URL of the MC server icon. How to set it: https://tinyurl.com/iconurl
    site: 'xxx', // URL of the MC server or vote website. Leave blank to disable site commands.
  },

  // Bot settings.
  settings: {
    language: {
      // Available languages: en (English), es (Spanish), de (German), fr (French), pt (Portuguese), ru (Russian), uk (Ukrainian)
      main: 'de', // Main language (files in ./translation/)
      // Optional language settings for specific features. Leave blank to use the main language.
      embeds: 'en', // Language for embeds, Slash and Prefix Commands, Auto Changing Status.
      autoReply: '', // Language for auto-reply feature responses.
      consoleLog: 'en', // Language for console log output.
      slashCmds: 'en', // Language for slash commands descriptions and error messages.
    },
    embedsColors: {
      basicCmds: 'Purple', // Color for basic commands like version, site, ip.
      online: 'Green', // Color for commands when the server is online (e.g., status, players, motd).
      offline: 'Red', // Color for offline status embeds.
    },
    // Console logging settings.
    logging: {
      timezone: 'Europe/Berlin', // Time zone for the bot. Use formats like America/New_York or Europe/London. Leave blank to use the bot's local time zone.
      inviteLink: true, // Log the invite link at the bot's launch.
      debug: false, // Log status messages and bot activity updates (may result in spam).
      error: true, // Log any errors that occur.
      serverInfo: true, // Log basic server info and check if it's online at startup.
    },
  },

  // Feature settings

  // Automatically updates the MC server status in a channel in real-time.
  autoChangeStatus: {
    enabled: true,
    updateInterval: 60, // Interval between status updates in seconds. Recommended: above 60.
    adminOnly: true, // Only admins with the "Manage Channel" permission can set the status message.
    // Show player head emoji in the player list. Only for Java and in adminOnly mode.
    playerAvatarEmoji: {
      enabled: true,
      guildID: 'xxx', // Server ID for creating/deleting emojis.
      // Consider creating a dedicated emoji Discord server and inviting the bot, or using the same server for status messages.
    },
    // These settings apply to slash (/) and prefix commands for status.
    isOnlineCheck: true, // Useful for servers using free hosting providers like Aternos. If the server's max players is 0, the status will be set to offline.
  },

  // Shows the player count of the MC server in the channel name.
  playerCountCH: {
    enabled: true,
    guildID: 'xxx', // Server ID for creating/editing channel stats.
    channelId: 'xxx', // Channel ID for editing the player count. If no ID is provided, the bot will create the channel itself.
    // {playeronline} and {playermax} display the current and maximum number of players.
    onlineText: 'ߟ⠻playeronline}/{playermax} active players',
    offlineText: 'ߔ䠏ffline', // Name set when the MC server is offline.
  },

  autoReply: {
    // If a message contains trigger words, reply with appropriate server information.
    enabled: false, // Disable the entire auto-reply feature.
    version: {
      enabled: true,
      triggerWords: ['version of the server?', 'version'],
    },
    ip: {
      enabled: true,
      triggerWords: ['ip of the server', 'ip'],
    },
    site: {
      enabled: true,
      triggerWords: ['website link', 'website', 'url', 'site', 'vote url', 'link'],
    },
    status: {
      enabled: true,
      triggerWords: ['is server online?', 'is server offline', 'status of the server'],
    },
  },

  commands: {
    slashCommands: true, // Enables all slash commands.
    prefixCommands: {
      enabled: true, // Enables all prefix commands.
      prefix: '!', // Prefix for normal commands.
    },
    ip: {
      enabled: true, // Enables the IP command.
      alias: ['ip-address'], // Aliases for IP prefix commands.
    },
    site: {
      enabled: true, // Enables the site command.
      alias: ['vote', 'link'], // Aliases for site prefix commands.
    },
    version: {
      enabled: true, // Enables the version command.
      alias: [], // Aliases for version prefix commands.
    },
    players: {
      enabled: true, // Enables the players command.
      alias: ['plist'], // Aliases for players prefix commands.
    },
    status: {
      enabled: true, // Enables the status command.
      alias: [], // Aliases for status prefix commands.
    },
    motd: {
      enabled: true, // Enables the motd command.
      alias: [], // Aliases for motd prefix commands.
    },
    help: {
      enabled: true, // Enables the help command.
      alias: ['commands'], // Aliases for help prefix commands.
    },
  },
}`

**Comment by NooberPro** [comment]
Thanks for sharing the config! I didn’t spot any issues there, so could you send over the logs and also the MC server IP and port? I’ll take a look from my end.

Oh, and try disabling playerAvatarEmojis—sometimes that helps clear things up.

Let me know!

**Comment by aeonoea** [comment]
Hey, sorry it took me a while to get back into this. So the latest log file I found in /root/.npm/logs seems unsuspicious:

0 verbose cli /usr/local/bin/node /usr/local/bin/npm
1 info using npm@10.7.0
2 info using node@v18.20.4
3 silly config:load:file:/usr/local/lib/node_modules/npm/npmrc
4 silly config:load:file:/app/.npmrc
5 silly config:load:file:/root/.npmrc
6 silly config:load:file:/usr/local/etc/npmrc
7 verbose title npm start
8 verbose argv "start"
9 verbose logfile logs-max:10 dir:/root/.npm/_logs/2024-09-12T14_34_23_647Z-
10 verbose logfile /root/.npm/_logs/2024-09-12T14_34_23_647Z-debug-0.log
11 silly logfile start cleaning logs, removing 1 files
12 silly logfile done cleaning log files

The minecraft server we're using is running on 178.39.113.4 and port 25568.

Let me know if you need more info for debugging, or which variables to set in config.js to log the needed info.

**Comment by aeonoea** [comment]
I corrected the IP in my previous comment, there was one number missing :)

**Comment by NooberPro** [comment]
Hey! Looks like the issue is with `playerAvatarEmoji`. Try setting `playerAvatarEmoji = false` in your config.js and see if that fixes it. This feature is new and might have some bugs, but it'll be sorted in the next update.

**Comment by NooberPro** [comment]
Did you invite the bot using the link given in console? Then it is not a permissions issue.
If not give your bot `administrator ` permissions so that it can edit and create channels. 
Don't put a channel id in the playerCountCH section and let the bot create a new channel.
Turn on the debug option in logging section in conifg file. And see if there are any error in the "console" of bot.
Also i recommend to download the bot again and reconfigure it as it may remove some accidental changes you may done.

**Comment by aeonoea** [comment]
Happy to report that '/setstatus' now is finally auto updating!

I kicked the bot, went to the developer settings and then chose 'bot' as scope and gave it 'administrator' permission (only) as you said. And then re-added the bot through the link given on the developer website.

Thank you so much for your help. Let us know how to buy you a coffee! :)

Btw, the bot is running in a docker [container](https://hub.docker.com/repository/docker/aeonoea/minecraftdiscordbot/general
) I created. If others want to access it too feel free to put the link somewhere. It's for amd64 systems only at the moment until I figured out how to compile for arm64 as well.
